java/src/com/google/idea/blaze/java/fastbuild/FastBuildChangedFilesService.java

```diff
@@ -356,5 +399,17 @@ void updateChangedSources(Set<File> changedFiles) {
         state = State.TOO_MANY_CHANGES;
       }
     }
+
+    void checkForNonCompilableChanges(Set<File> changedNonCompileFiles) {
```

Comment:
why separate this logic from updatechangedsources? Should this be in the same method?

Reply:
> The job of updateChangedSources is to take the changed files, filter the relevant ones from the sources, and adding them to changedSources property, which is the property, that collects the files that get compiled when getAndResetChangedSources.

Why does this not apply to protos? Even if we stay with this approach then it's worthwhile explaining why

Reply:
Because with Java/Scala files, the changed files need to be collected, as they are the files that get compiled. In case of proto, you don't really need to track the files, you need to only know, that any changed proto files exist